tests/h/models/user_identity_test.py

```diff
@@ -43,29 +45,44 @@ def test_you_can_save_and_then_retrieve_field_values(self, db_session, matchers)
         assert user_identities[1].provider_unique_id == "2"
         assert user_identities[2].provider_unique_id == "3"
 
-    def test_provider_cant_be_null(self, db_session):
-        db_session.add(models.UserIdentity(provider_unique_id="1"))
+    def test_provider_cant_be_null(self, db_session, user):
+        db_session.add(models.UserIdentity(provider_unique_id="1", user=user))
 
         with pytest.raises(
             sqlalchemy.exc.IntegrityError,
             match='null value in column "provider" violates not-null constraint',
         ):
             db_session.flush()
 
-    def test_provider_id_cant_be_null(self, db_session):
-        db_session.add(models.UserIdentity(provider="provider"))
+    def test_provider_id_cant_be_null(self, db_session, user):
+        db_session.add(models.UserIdentity(provider="provider", user=user))
 
         with pytest.raises(
             sqlalchemy.exc.IntegrityError,
             match='null value in column "provider_unique_id" violates not-null constraint',
         ):
             db_session.flush()
 
-    def test_two_cant_have_the_same_provider_and_provider_id(self, db_session):
+    def test_user_cant_be_null(self, db_session):
+        db_session.add(models.UserIdentity(provider="provider", provider_unique_id="1"))
+
+        with pytest.raises(
+            sqlalchemy.exc.IntegrityError,
+            match='null value in column "user_id" violates not-null constraint',
+        ):
+            db_session.flush()
+
+    def test_two_cant_have_the_same_provider_and_provider_id(
+        self, db_session, factories
+    ):
         db_session.add_all(
             [
-                models.UserIdentity(provider="provider", provider_unique_id="id"),
-                models.UserIdentity(provider="provider", provider_unique_id="id"),
+                models.UserIdentity(
+                    provider="provider", provider_unique_id="id", user=factories.User()
```

Comment:
This test might be a little more controlled if both `UserIdentity` models made use of the same `User` instead of generating a new one each time?

Reply:
I actually want them to be two different users - ensuring that even if the user is different, the two `UserIdentity`'s still aren't allowed if they have the same `provider` and `provider_id`